crypto/rand/rand_unix.c

```diff
@@ -256,5 +285,113 @@ size_t rand_pool_acquire_entropy(RAND_POOL *pool)
 #  endif
 }
 # endif
+#endif
+
+#ifdef OPENSSL_SYS_UNIX
+int rand_pool_add_nonce_data(RAND_POOL *pool)
+{
+    struct {
+        pid_t pid;
+        CRYPTO_THREAD_ID tid;
+        uint64_t time;
+    } data = { 0 };
+
+    /*
+     * Add process id, thread id, and a high resolution timestamp to
+     * ensure that the nonce is unique whith high probability for
+     * different process instances.
+     */
+    data.pid = getpid();
+    data.tid = CRYPTO_THREAD_get_current_id();
+    data.time = get_time64();
```

Comment:
This would be good for additional data, not the nonce.

Reply:
I totally agree with you. The perfect nonce ingredient would be a 64 bit high resolution utc timestamp (like it's the case for Windows), because that decreases the overall collision probability drastically, provided the computer's clocks are synchronized properly. Unfortunately, it seems to be almost impossible to get that for all unix-ish platforms in a portable way. I'm sure you remember the struggle we had with the timers in #5231. I'm also not happy with the `#ifdef` spaghetti in `get_time64()`, but it seems we had no other choice. So if you have any suggestions how to improve that, I would be happy to hear about it.

Reply:
One could use different timers for nonce and additional data, the former optimized for uniqueness and the latter for speed, Since the noise is in the lower bits, OPENSSL_rdtsc() would be sufficient for additional data.

Reply:
There are things like gettimeofday() and time(), which I think should be supported on all POSIX platforms.

Reply:
Note that I don't see 64 bit as a requirement, gettimeofday() should be good enough.